differ/differ.go

```diff
@@ -294,7 +302,183 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
 	return evaluator.Evaluate(eventFilter, values)
 }
 
-func processReportsByCluster(ruleContent types.RulesMap, storage Storage, clusters []types.ClusterEntry) {
+// ccx_rules_ocp.external.rules.cluster_wide_proxy_auth_check
+// ->
+// cluster_wide_proxy_auth_check
+func ruleIDToRuleName(ruleID types.RuleID) types.RuleName {
+	result := strings.TrimPrefix(string(ruleID), "ccx_rules_ocp.")
+	result = strings.TrimPrefix(result, "external.")
+	result = strings.TrimPrefix(result, "rules.")
+	result = strings.TrimPrefix(result, "bug_rules.")
+	result = strings.TrimPrefix(result, "ocs.")
+
+	return types.RuleName(result)
+}
+
+func findRenderedReport(reports []types.RenderedReport, ruleName types.RuleName, errorKey types.ErrorKey) (types.RenderedReport, error) {
+	for _, report := range reports {
+		if ruleIDToRuleName(report.RuleID) == ruleName && report.ErrorKey == errorKey {
+			return report, nil
+		}
+	}
+	return types.RenderedReport{}, errors.New("report for given rule ID nad error key has not been found")
+}
+
+func produceEntriesToServiceLog(config conf.ConfigStruct, cluster types.ClusterEntry,
+	ruleContent types.RulesMap, reports []types.ReportItem) {
+	renderedReports, err := renderReportsForCluster(
+		conf.GetDependenciesConfiguration(config), cluster.ClusterName,
+		reports, ruleContent)
+	if err != nil {
+		log.Err(err).
+			Str("cluster name", string(cluster.ClusterName)).
+			Msg("Rendering reports failed for this cluster")
+		return
+	}
+	for _, r := range reports {
+		module := r.Module
+		ruleName := moduleToRuleName(module)
+		errorKey := r.ErrorKey
+
+		likelihood, impact, totalRisk, _ := findRuleByNameAndErrorKey(ruleContent, ruleName, errorKey)
+		eventValue := EventValue{
+			Likelihood: likelihood,
+			Impact:     impact,
+			TotalRisk:  totalRisk,
+		}
+
+		// try to evaluate event filter expression
+		result, err := evaluateFilterExpression(serviceLogEventFilter,
+			serviceLogEventThresholds, eventValue)
+		if err != nil {
+			log.Err(err).Msg(evaluationErrorMessage)
+			continue
+		}
+
+		if result > 0 {
+			renderedReport, err := findRenderedReport(renderedReports, ruleName, errorKey)
+			if err != nil {
+				log.Err(err).Msgf("Output from content template renderer does not contain "+
+					"result for cluster %s, rule %s and error key %s", cluster.ClusterName, ruleName, errorKey)
+				continue
+			}
+
+			logEntry := types.ServiceLogEntry{
+				ClusterUUID: string(cluster.ClusterName),
+				Description: renderedReport.Description,
+				ServiceName: "CCX Notification Service",
+				Summary:     renderedReport.Reason,
+			}
+
+			// It is necessary to truncate the fields because of Service Log limitations
+			if len(logEntry.Description) > 255 {
```

Comment:
probably better to call named function (refactoring) + named constant instead of magic constant

Reply:
Done